Paintroid/src/main/java/org/catrobat/paintroid/tools/helper/FillAlgorithm.java

```diff
@@ -53,6 +55,7 @@ public FillAlgorithm(Bitmap bitmap, Point clickedPixel, int targetColor, int rep
 		mTargetColor = targetColor;
 		mReplacementColor = replacementColor;
 		mRanges = new LinkedList<Range>();
+		mColorToleranceThreshold = colorToleranceThreshold;
```

Comment:
As explained below in my comment at line 142, replace by

		mColorToleranceThresholdSquared = Math.pow(colorToleranceThreshold, 2);